pkg/pillar/cmd/nodeagent/handletimers.go

```diff
@@ -203,16 +203,60 @@ func scheduleNodeReboot(ctxPtr *nodeagentContext, reasonStr string) {
 
 	// in any case, execute the reboot procedure
 	// with a delayed timer
-	go handleNodeReboot(reasonStr)
+	go handleNodeReboot(ctxPtr, reasonStr)
 }
```

Comment:
If the domain failed to boot will it be in halted state? If not you can check the activated Boolean instead.

Reply:
good point.. Checking only For Halted is indeed a slippery slope..But the code looked like it was ok to make the check.. But Checking for Activated seems safer..

Made the change..

Reply:
Activated          bool    // XXX remove??

There is this comment.. Will remove the comment too..